aten/src/ATen/native/UpSample.h

```diff
@@ -0,0 +1,249 @@
+#include <math.h>
+
+#include <ATen/ATen.h>
+
+namespace at {
+namespace native {
+
+static inline void check_dim_size(
+    const Tensor& data,
+    int64_t dim,
+    int64_t dim_size,
+    int64_t size) {
+  /* Check dimension size of a tensor */
+  AT_CHECK(
+      data.dim() == dim && data.size(dim_size) == size,
+      "Expected tensor of dimension ",
+      dim,
+      " and tensor.size[",
+      dim_size,
+      "] == ",
+      size,
+      " but got: dimension ",
+      data.dim(),
+      " and tensor.size[",
+      dim_size,
+      "] = ",
+      data.size(dim_size));
+}
+
+static inline void upsample_1d_shape_check(
+    const Tensor& data,
+    int64_t type_check,
```

Comment:
Um. Your heart is in the right place, but you can't just put an undocumented `int64_t` and not say what the values for it mean. At the *very* least you should have defined an enum class; but, I probably would have done a more faithful translation, replacing null pointer with an "undefined" tensor and then testing (`input.defined()`).

Reply:
> but, I probably would have done a more faithful translation, replacing null pointer with an "undefined" tensor and then testing (input.defined()).

yes, that would be awesome!

should I wait for that? or just use an enum class?

Reply:
or should do it? if yes, could you give some guidance?

Reply:
but maybe would be good to add this in another PR.

Reply:
Undefined tensor is already supported. When you say `Tensor()` that is an undefined tensor; similar `Tensor x;` x is undefined. Nothing to wait for, you just need to fix your code.

Reply:
oh sorry .. I misunderstood ... I will use that. thanks!